File: fixtures/scale/python/large/cipherlab/checks/check_dualmask.py
from cipherlab.ciphers.dualmask import dualmask_encrypt, dualmask_decrypt
from cipherlab.specs.dualmask_spec import dualmask_spec
from cipherlab.corpus.corpus_mask import corpus_mask


def check_dualmask():
    spec = dualmask_spec()
    for text in corpus_mask():
        sealed = dualmask_encrypt(text, spec["key"])
        opened = dualmask_decrypt(sealed, spec["key"])
        if opened != text:
            return False
    return spec["category"] == "mask"

File: fixtures/scale/ruby/large/lib/check_dualmask.rb
require_relative 'dualmask_encrypt'
require_relative 'dualmask_decrypt'
require_relative 'corpus_mask'

def check_dualmask
  corpus_mask.each do |text|
    sealed = dualmask_encrypt(text, 5)
    opened = dualmask_decrypt(sealed, 5)
    return false if opened != text
  end
  true
end
